src/main/java/com/imsweb/mph/MphUtils.java

```diff
@@ -238,6 +238,7 @@ else if (!group1.getId().equals(group2.getId())) {
             output.setReason("The two sets of parameters belong to two different cancer groups.");
         }
         else {
+            output.setGroupId(group1.getId());
```

Comment:
This code looks suspicious to me.

The ID and the name are just to representation of a group.  They should never be out of sync.

You assign the ID here based on "group1". But you assign the name later in the logic, based on the one of the rules.

I am not saying this is an actual problem, but if the ID and name can never get out of sync, I would prefer to see them being assign together using the same object.

It's OK if you want to assign both here. It's also OK if you want to assign both based on the rule later in the logic. But I think they should be assigned together.